shader/antialiasing/cmaa2/cmaa2_common.glsl

```diff
@@ -0,0 +1,134 @@
+#define CMAA_PACK_SINGLE_SAMPLE_EDGE_TO_HALF_WIDTH  1
+#define CMAA2_PROCESS_CANDIDATES_NUM_THREADS        128
+#define CMAA2_DEFERRED_APPLY_NUM_THREADS            32
+#define CMAA2_DEFERRED_APPLY_THREADGROUP_SWAP       1 
+#define CMAA2_USE_HALF_FLOAT_PRECISION              0
+#define CMAA2_SUPPORT_HDR_COLOR_RANGE               0
+#define CMAA2_EDGE_DETECTION_LUMA_PATH              0       // We should use HDR luma from a separate buffer in the future
+#define CMAA2_EDGE_UNORM                            0
+#define CMAA2_EXTRA_SHARPNESS                       0
+
+const float symmetryCorrectionOffset = 0.22;
+
+#if CMAA2_EXTRA_SHARPNESS
+  const float dampeningEffect = 0.11;
+  const float cmaa2LocalContrastAdaptationAmount = 0.15f
+  const float cmaa2SimpleShapeBlurinessAmount = 0.07f
+#else
+  const float dampeningEffect = 0.15;
+  const float cmaa2LocalContrastAdaptationAmount =  0.10f;
+  const float cmaa2SimpleShapeBlurinessAmount = 0.10f;
+#endif
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// VARIOUS QUALITY SETTINGS
+//
+// Longest line search distance; must be even number; for high perf low quality start from ~32 - the bigger the number, 
+// the nicer the gradients but more costly. Max supported is 128!
+const uint maxLineLength = 86;
+// 
+
+layout(binding = 0) uniform sampler2D inputColorReadonly;
+
+#if CMAA2_EDGE_UNORM 
+  layout(r8, binding = 2) uniform image2D workingEdges;
+#else
+  layout(r32ui, binding = 2) uniform uimage2D workingEdges;
+#endif
+
+layout(binding = 3) buffer UboWorkingShapeCandidates {
+  uint workingShapeCandidates[];
+  };
+
+layout(binding = 4) buffer UboWorkingDeferredBlendLocationList {
+  uint workingDeferredBlendLocationList[];
+  };
+
+layout(binding = 5) buffer UboWorkingDeferredBlendItemList {
+  uvec2 workingDeferredBlendItemList[];
+  };
+
+layout(r32ui, binding = 6) uniform uimage2D workingDeferredBlendItemListHeads;
+
+layout(binding = 7) buffer UboWorkingControlBuffer {
+  uint workingControlBuffer[];
```

Comment:
No, extra variants didnt require additional counters. I suppose counters are aligned for faster access. 3 and 4 are close probably because they are used together in the indirect setup pass

Reply:
There is a publication about atomics performance profile: https://www.youtube.com/watch?v=VaE_uKPfjv0
I don't remember details now, yet my takeaway was that only single variable case is fast-path. Anything else - unpredictable across multiple gpu vendors.

In any case no point of using array, if you really need padding - use struct:
```c++
buffer {
  uint numCandidates;
  uint shapeCandidateCount;
  uint padding0[3];
  uint blendLocationCount;
  ...
  }
```